src/exchange_adapters/alphavantage.ts

```diff
@@ -0,0 +1,142 @@
+import { BaseExchangeAdapter, ExchangeAdapter, ExchangeDataType, Ticker } from './base'
+
+import BigNumber from 'bignumber.js'
+import { Exchange } from '../utils'
+import { strict as assert } from 'assert'
+
+enum ResponseKeys {
+  fromCurrency = '1. From_Currency Code',
+  toCurrency = '3. To_Currency Code',
+  rate = '5. Exchange Rate',
+  lastUpdated = '6. Last Refreshed',
+  bid = '8. Bid Price',
+  ask = '9. Ask Price',
+}
+
+type ResponseData = {
+  [K in ResponseKeys]: string
+}
+
+interface Response {
+  'Realtime Currency Exchange Rate': ResponseData
+}
+
+export class AlphavantageAdapter extends BaseExchangeAdapter implements ExchangeAdapter {
+  baseApiUrl = 'https://www.alphavantage.co'
+  readonly _exchangeName: Exchange = Exchange.ALPHAVANTAGE
+  readonly _certFingerprint256 =
+    '3A:BB:E6:3D:AF:75:6C:50:16:B6:B8:5F:52:01:5F:D8:E8:AC:BE:27:7C:50:87:B1:27:A6:05:63:A8:41:ED:8A'
+
+  protected generatePairSymbol(): string {
+    const base = AlphavantageAdapter.standardTokenSymbolMap.get(this.config.baseCurrency)
+    const quote = AlphavantageAdapter.standardTokenSymbolMap.get(this.config.quoteCurrency)
+
+    return `${base}${quote}`
```

Comment:
wouldn't it make sense to return something like 
`from_currency=${base}&to_currency=${quote}`
and then use this.pairSymbol in fetchTicker and is OrderBookLive?

Reply:
I see what you mean but I think it would be a bit strange since it's not really a symbol. For this adapter this function doesn't make a lot of sense but I had to implement it anyways

Reply:
ok makes sense